package.json

```diff
@@ -28,7 +28,8 @@
     "test": "yarn build && yarn test:post-build",
     "test:post-build": "node dist/index.js test",
     "start": "tsc -p tsconfig.json --watch",
-    "release": "np"
+    "release": "np",
+    "deduplicate": "npx yarn-deduplicate -s fewer yarn.lock"
```

Comment:
This should probably be a `postinstall` hook (and at the top of scripts)

Reply:
After running that dedup, you need to run yarn install again to make the yarn.lock changes to actually effect the node_modules
And adding another yarn install as post install hook can make a loop and mess

My personal way of using this tool is mostly manually

Reply:
> The `--list --fail` exits, but it wasn't documented

> I've made similar PR to webpack, this is how they eventually choose to impl it:
> https://github.com/webpack/webpack/blob/master/package.json#L157-L158
> `yarn-lint` runs as part of general lint
> `yarn-lint-fix` is manual

Ok will try this out, thanks for the notes. I was asking more of if that might cause a loop but guess not

> I personally don't really like pre commit hooks, (Not for failing or for mutating source code like prettier fix)
> I prefer CI checks and pre push

Yea I get that it can be a bit of a hassle when you've got a commit in your head or no relevant files have changed and then you have to wait for pre-commit to run. But with pre-push only and CI, you miss potential errors in between commits that are fixed before the final push/PR, meaning it allows for individual commits to be invalid so long as the whole is valid.

For the most part, PRs get squash-merged, but sometimes I rebase things that are meant to be distinct commits